### PR TITLE
[C-2465] Improve audius-query cache selection performance

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Improves query cache performance by checking denormalized result with `isEqual`. This is a temporary fix until we can update `denormalize` function to properly select from cache.

### CODE REVIEW

File: packages/common/src/audius-query/createApi.ts [hunk 1]
```diff
@@ -143,7 +142,7 @@ const useQueryState = <Args, Data>(
   reducerPath: string,
   endpointName: string,
   endpoint: EndpointConfig<Args, Data>
-) => {
+): Nullable<PerKeyState<any> & { isInitialValue?: boolean }> => {
```
Comment (amendelsohn) [inline packages/common/src/audius-query/createApi.ts:145]: does `PerKeyState<Data>` work? Maybe not cause it's like `PerKeyState<Normalized<Data>` really

File: packages/common/tsconfig.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "outDir": "dist",
     "baseUrl": ".",
     "paths": {
+      "audius-query": ["src/audius-query/index.ts"],
```
Comment (amendelsohn) [inline packages/common/tsconfig.json:6]: what does this do?

**Review by amendelsohn — APPROVED** [review]
So glad this works!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-fix-audius-query-cache-performance

File: packages/common/src/audius-query/createApi.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/common/src/audius-query/createApi.ts:145]: Ah i see i can try that, the main thing was complaining about isInitialValue not being a port of PerStateKey

File: packages/common/tsconfig.json (hunk 2 above)
Comment (dylanjeffers) [reply packages/common/tsconfig.json:6]: makes absolute imports possible, which i kinda like with audius-query cause it's kinda like it's own package

Comment (amendelsohn) [reply packages/common/tsconfig.json:6]: OH that's cool

**Pull request merged**